examples/pysdl2.py

```diff
@@ -0,0 +1,182 @@
+#
+# Requires pysdl2 (and SDL2 library).
+#
+# Tested under Fedora Linux (x86_64).
+#
+# by Neil Munday (www.mundayweb.com)
+#
+
+import os
+import sys
+import sdl2
+import sdl2.ext
+
+from cefpython3 import cefpython as cef
+from PIL import Image
+
+class LoadHandler(object):
+	def OnLoadingStateChange(self, browser, is_loading, **_):
+		if not is_loading:
+			print "loading complete"
+			
+	def OnLoadError(self, browser, frame, error_code, failed_url, **_):
+		if not frame.IsMain():
+			return
+		print "Failed to load %s" % failed_url
+		cef.PostTask(cef.TID_UI, exit_app, browser)
+
+class RenderHandler(object):
+	
+	def __init__(self, renderer, width, height):
+		self.__width = width
+		self.__height = height
+		self.__renderer = renderer
+		self.texture = None
+			
+	def GetViewRect(self, rect_out, **_):
+		rect_out.extend([0, 0, self.__width, self.__height])
+		return True
+	
+	def GetScreenRect(self, browser, rect_out):  # noqa: N802
+		return False
+	
+	def GetScreenPoint(self, browser, view_x, view_y, screen_coordinates_out):
+		return False
+	
+	def OnPaint(self, browser, element_type, paint_buffer, **_):
+		#
+		# Use PIL to create a set of bytes that we can turn into an SDL2 surface
+		# and then convert this to a SDL2 texture for rendering the main program loop.
+		#
+		if element_type == cef.PET_VIEW:
+			data = paint_buffer.GetString(mode="rgba", origin="top-left")
+			image = Image.frombuffer('RGBA', (self.__width, self.__height), data, 'raw', 'BGRA')
+			#
+			# Following PIL to SDL2 surface code from pysdl2 source
+			#
+			mode = image.mode
+			rmask = gmask = bmask = amask = 0
+			if mode == "RGB":
+				# 3x8-bit, 24bpp
+				if sdl2.endian.SDL_BYTEORDER == sdl2.endian.SDL_LIL_ENDIAN:
+					rmask = 0x0000FF
+					gmask = 0x00FF00
+					bmask = 0xFF0000
+				else:
+					rmask = 0xFF0000
+					gmask = 0x00FF00
+					bmask = 0x0000FF
+				depth = 24
+				pitch = self.__width * 3
+			elif mode in ("RGBA", "RGBX"):
+				# RGBX: 4x8-bit, no alpha
+				# RGBA: 4x8-bit, alpha
+				if sdl2.endian.SDL_BYTEORDER == sdl2.endian.SDL_LIL_ENDIAN:
+					rmask = 0x00000000
+					gmask = 0x0000FF00
+					bmask = 0x00FF0000
+					if mode == "RGBA":
+						amask = 0xFF000000
+				else:
+					rmask = 0xFF000000
+					gmask = 0x00FF0000
+					bmask = 0x0000FF00
+					if mode == "RGBA":
+						amask = 0x000000FF
+				depth = 32
+				pitch = self.__width * 4
+			else:
+				print "Unsupported mode: %s" % mode
+				exit_app()
+			
+			pxbuf = image.tobytes()
+			# create surface
+			surface = sdl2.SDL_CreateRGBSurfaceFrom(pxbuf, self.__width, self.__height, depth, pitch, rmask, gmask, bmask, amask)
+			
+			if self.texture:
+				sdl2.SDL_DestroyTexture(self.texture)
+			# create texture
+			self.texture = sdl2.SDL_CreateTextureFromSurface(self.__renderer, surface)
+			sdl2.SDL_FreeSurface(surface)
+		else:
+			print "Unsupport element_type in OnPaint"
+
+def exit_app():
+	sdl2.SDL_Quit()
+	cef.Shutdown()
+	print "exited"
+
+if __name__ == "__main__":
```

Comment:
Please put here only a call to main(), and define main at the top of the file just below the imports. That's how it is done in all other examples, so it should be consistent in this example as well.